docs/parameters.md

```diff
@@ -0,0 +1,150 @@
+ * The user has deployed an external database configured with TLS and needs to specify the CA certificate.
+ * The user has deployed an external database configured with TLS and certificate authentication.
+
+##### Mapped
+
+| Parameter | Description | foreman-installer Parameters |
+| ----------| ----------- | ---------------------------- |
+| `--database-host` | Location to connect to the database | `--foreman-db-host`<br/> `--katello-candlepin-db-host`<br/> `--foreman-proxy-content-pulpcore-postgresql-host` |
+| `--database-port` | Port to connect to the database | `--foreman-db-port`<br/> `--katello-candlepin-db-port`<br/> `--foreman-proxy-content-pulpcore-postgresql-port` |
+| `--database-password` | Password for each database | `--foreman-db-password`<br/> `--katello-candlepin-db-password`<br/> `--foreman-proxy-content-pulpcore-postgresql-password` |
+| `--database-ssl-mode` | SSL verification mode to use | `--foreman-db-sslmode` <br/> `--katello-candlepin-db-ssl-verify` |
+| `--database-ssl-ca` | Path to the database CA certificate | `--foreman-db-root-cert` <br/> `--katello-candlepin-db-ssl-ca` <br/> `--foreman-proxy-content-pulpcore-db-ssl-root-ca` |
+| `--database-ssl-cert` | Path to the database certificate | |
+| `--database-ssl-key` | Path to the database private key | |
+
+##### Unmapped
+
+| foreman-installer Parameter | Description | Reason |
+| --------------------------- | ----------- | ------ |
+| `--katello-candlepin-manage-db` | When true, creates a local database | ?? |
+| `--katello-candlepin-db-name` | Sets the name of the Candlepin database | The user should not configure this and instead use a hard-coded name |
+| `--katello-candlepin-db-user` | Sets the user of the Candlepin database | The user should not configure this and instead use a hard-coded name |
+| `--katello-candlepin-db-ssl` | Sets whether database connection is over SSL | The SSL mode parameter can be used to determine this |
+| `--foreman-proxy-content-pulpcore-manage-postgresql` | When true, creates a local database | ?? |
+| `--foreman-proxy-content-pulpcore-postgresql-db-name` | Sets the name of the Pulpcore database | The user should not configure this and instead use a hard-coded name |
+| `--foreman-proxy-content-pulpcore-postgresql-user` | Sets the user of the Pulpcore database | The user should not configure this and instead use a hard-coded name |
+| `--foreman-proxy-content-pulpcore-postgresql-ssl` | Sets whether database connection is over SSL | The SSL mode parameter can be used to determine this |
+| `--foreman-db-manage` | When true, creates a local database | ?? |
+| `--foreman-db-database` | Sets the user of the Foreman database | The user should not configure this and instead use a hard-coded name |
+| `--foreman-db-username` | Sets the user of the Foreman database | The user should not configure this and instead use a hard-coded name |
```

Comment:
I know of users (of ext db), who use that, which will make a migration for them harder.

Also, I can imagine, when we allow "run whatever you want as long it's postgesql version X compatible", we'll have to deal with weird db names and user names because *some* managed platform will fuck it up for us.

Reply:
Our documentation explicitly tells users to create the username and database to our specifications:

```
CREATE USER "foreman" WITH PASSWORD 'Foreman_Password';
CREATE USER "candlepin" WITH PASSWORD 'Candlepin_Password';
CREATE USER "pulp" WITH PASSWORD 'Pulpcore_Password';
CREATE DATABASE foreman OWNER foreman;
CREATE DATABASE candlepin OWNER candlepin;
CREATE DATABASE pulpcore OWNER pulp;
```

Reply:
This should add the following as input parameters:

- Each database user
- Each database name
- Each database password